experimental/uefi/app/src/serial.rs

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use uefi::{
+    prelude::*,
+    proto::console::serial,
+    table::boot::{OpenProtocolAttributes, OpenProtocolParams},
+    Error, Handle, ResultExt, Status,
+};
+
+pub struct Serial<'boot> {
+    serial: &'boot mut serial::Serial<'boot>,
+}
+
+impl<'boot> Serial<'boot> {
+    pub fn get(handle: Handle, bt: &BootServices, index: usize) -> Result<Serial, Error<()>> {
+        // Expect (at least) two serial ports on the system; the first will be used
+        // for stdio, and we can use the second one for our echo example. If we don't
```

Comment:
Why do we need a separate port for the example? Can messages be transfered through an stdio channel?

Reply:
In principle we could channel everything over stdio. However, given that stdio also gets all the log output etc, having a separate serial port that isn't used by anyone but us makes transferring data between the loader and the app much simpler.